frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,224 @@
 import { HardwareProfileKind } from '~/k8sTypes';
+import { Identifier, IdentifierResourceType } from '~/types';
+import { determineUnit, splitValueUnit } from '~/utilities/valueUnits';
+import { hasCPUandMemory } from './manage/ManageNodeResourceSection';
+
+export type WarningNotification = {
+  warningStatus: boolean;
+  title: string;
+  message: string;
+  isDefaultProfile: boolean;
```

Comment:
This prop `isDefaultProfile` is also not very useful here. It's only used to decide whether to show a button to restore default or not when rendering the popover, you can just do `isHardwareProfileOOTB(profile)` there in the table row. No need to pass it around and make the type complicated. Keep it as simple as possible (P.S. `isDefaultProfile` should not be in `warning`, when structuring a type, we also need to consider the context and see whether we really need it or if it's reasonable enough to be here, just a suggestion)